presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -607,16 +607,34 @@ void PrestoServer::initializeVeloxMemory() {
   const uint64_t memoryGb = systemConfig->systemMemoryGb();
   PRESTO_STARTUP_LOG(INFO) << "Starting with node memory " << memoryGb << "GB";
 
-  const int64_t memoryBytes = memoryGb << 30;
+  // Set up velox memory manager.
+  memory::MemoryManagerOptions options;
+  options.allocatorCapacity = memoryGb << 30;
   if (systemConfig->useMmapAllocator()) {
-    memory::MmapAllocator::Options options;
-    options.capacity = memoryBytes;
+    options.useMmapAllocator = true;
     options.useMmapArena = systemConfig->useMmapArena();
     options.mmapArenaCapacityRatio = systemConfig->mmapArenaCapacityRatio();
-    allocator_ = std::make_shared<memory::MmapAllocator>(options);
-  } else {
-    allocator_ = memory::MemoryAllocator::createDefaultInstance();
   }
+  options.checkUsageLeak = systemConfig->enableMemoryLeakCheck();
+  options.trackDefaultUsage =
+      systemConfig->enableSystemMemoryPoolUsageTracking();
+  if (!systemConfig->memoryArbitratorKind().empty()) {
+    options.arbitratorKind = systemConfig->memoryArbitratorKind();
+    const uint64_t queryMemoryGb = systemConfig->queryMemoryGb();
+    VELOX_USER_CHECK_LE(
+        queryMemoryGb,
+        memoryGb,
+        "Query memory capacity must not be larger than system memory capacity");
+    options.arbitratorCapacity = queryMemoryGb << 30;
+    options.memoryPoolInitCapacity = systemConfig->memoryPoolInitCapacity();
+    options.memoryPoolTransferCapacity =
+        systemConfig->memoryPoolTransferCapacity();
+    options.arbitrationStateCheckCb = velox::exec::memoryArbitrationStateCheck;
+  }
+  memory::initializeMemoryManager(options);
+  PRESTO_STARTUP_LOG(INFO) << "Memory manager has been setup: "
```

Comment:
Nit : Maybe follow the message style "Memory manager {} has been setup", memory::memoryManager()->toString() as AsyncDataCache message.

Reply:
The memory manger toString has multiple lines so it is better to put them at the end of log message.